From git-commits-head-owner@vger.kernel.org Sat Sep 30 01:10:28 2006
Date: Fri, 29 Sep 2006 17:01:39 GMT
Message-Id: <200609291701.k8TH1dVv030631@hera.kernel.org>
From: Linux Kernel Mailing List <linux-kernel@vger.kernel.org>
To: git-commits-head@vger.kernel.org
Subject: [PATCH] use gcc -O1 in fs/reiserfs only for ancient gcc versions

commit 42012cc4a2183c555a907eee32d7ce4fc7dc3a6a
tree 90fbd712085050fac8f06544e3186736539cf2bc
parent 4d24607bfaab3d02fd0d6e0ffc22f29c2bc8eb30
author Olaf Hering <olh@suse.de> 1159520361 -0700
committer Linus Torvalds <torvalds@g5.osdl.org> 1159546687 -0700

[PATCH] use gcc -O1 in fs/reiserfs only for ancient gcc versions

Only compile with -O1 if the (very old) compiler is broken.  We use
reiserfs alot since SLES9 on ppc64, and it was never seen with gcc33.
Assume the broken gcc is gcc-3.4 or older.

Signed-off-by: Olaf Hering <olh@suse.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Linus Torvalds <torvalds@osdl.org>

 fs/reiserfs/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/reiserfs/Makefile b/fs/reiserfs/Makefile
index 3a59309..0eb7ac0 100644
--- a/fs/reiserfs/Makefile
+++ b/fs/reiserfs/Makefile
@@ -28,7 +28,7 @@ # and causing a panic. Since this behavi
 # will work around it. If any other architecture displays this behavior,
 # add it here.
 ifeq ($(CONFIG_PPC32),y)
-EXTRA_CFLAGS := -O1
+EXTRA_CFLAGS := $(call cc-ifversion, -lt, 0400, -O1)
 endif
 
 TAGS:

